src/components/form/SelectPositions/index.tsx

```diff
@@ -0,0 +1,201 @@
+import { positionString } from 'util/tools'
+import { Errors } from 'util/types'
+
+import { ButtonControl, ButtonControlType } from 'components/buttons/ButtonControl'
+import { SelectPositionModal } from 'components/modals/SelectPositionsModal'
+import { Error, ErrorContainer } from 'components/pureStyledComponents/Error'
+import {
+  StripedList,
+  StripedListEmpty,
+  StripedListItem,
+} from 'components/pureStyledComponents/StripedList'
+import { Textfield } from 'components/pureStyledComponents/Textfield'
+import { TitleControl } from 'components/pureStyledComponents/TitleControl'
+import { InlineLoading } from 'components/statusInfo/InlineLoading'
+import { TitleValue } from 'components/text/TitleValue'
+import { useBatchBalanceContext } from 'contexts/BatchBalanceContext'
+import { useMultiPositionsContext } from 'contexts/MultiPositionsContext'
+import { useWeb3ConnectedOrInfura } from 'contexts/Web3Context'
+import { BigNumber } from 'ethers/utils'
+import { Position } from 'hooks'
+import React from 'react'
+import { GetMultiPositions_positions } from 'types/generatedGQL'
+
+interface Props {
+  title: string
+  singlePosition?: boolean
+  showOnlyPositionsWithBalance?: boolean
+  callbackToBeExecutedOnRemoveAction?: () => void
+}
+
+const canCanculatePositionToDisplay = (
+  positionsLoading: boolean,
+  balancesLoading: boolean,
+  positions: GetMultiPositions_positions[],
+  balances: BigNumber[],
+  positionIds: string[]
+) => {
+  return (
+    !positionsLoading &&
+    !balancesLoading &&
+    positions.length &&
+    balances.length &&
+    balances.length === positionIds.length &&
+    positions.length === positionIds.length &&
+    JSON.stringify(positions.map(({ id }) => id).sort()) === JSON.stringify([...positionIds].sort())
```

Comment:
Mmm in your own words, what is the intention of this ?

Reply:
There is a scenario where you change selected positions for different group with the same number of positions, so this checks that all the values are in sync